Added Pkg and CI Template Creation

Joschi3 opened the pull request:
Allows the creation of cpp_pkgs, python_pkgs, and msgs_pkgs using a copier.
Additionally, you can create CI Templates for github and gitlab.

Usage:
-  `hector create ` to create pkgs directly in the workspace src folder
- `hector create -d <repository_name>` to create Pkgs / Continous Integration Pipelines in a repository in the src folder

Copier will lead you through a questionnaire, allowing you to select further options easily.
The author name and email are extracted from the global git config.

Joschi3 commented on scripts/create.py:
```diff
@@ -0,0 +1,259 @@
+try:
+    import copier
+except ImportError:
+    print("GitPython is required! Install using 'pip3 install --user copier'")
```
No. Unfortunately, there is no apt package ...

StefanFabian replied:
rosdep should also support pip packages, might make sense to check if there is already a key and if not make a PR so it can be listed in the dependencies of the package and installed automatically.

Joschi3 replied:
I have created a [pull request](https://github.com/ros/rosdistro/pull/44327#pullrequestreview-2586365756) for adding copier to rosdep.

StefanFabian commented on scripts/create.py:
```diff
@@ -0,0 +1,220 @@
+    import copier
+except ImportError:
+    print(
+        "Copier is required! Install using 'pip3 install copier --break-system-packages'"
```
Why did you drop the `--user`?

StefanFabian commented on scripts/create.py:
```diff
@@ -0,0 +1,220 @@
+        help="CI type",
+    )
+    parser.add_argument(
+        "--add_pre_commit",
```
This is the only argument using _ in its name.

StefanFabian commented:
This always creates the package in the src folder, would it not make more sense to default to the current working directory?

StefanFabian commented:
Also the launch file it creates has a namespace arg which doesn't make sense for us since we extended ros2 launch to allow specifying a namespace which is our preferred less hacky way.
I also find Add parameter loading a bit misleading since it adds dynamic parameter loading not just loading it once.
That's really neat (though Martin and I both think it might make sense to create an extended Node base class handling this instead of repeating this code everywhere) but I only needed a startup parameter.

Joschi3 commented:
I’ve made the following changes based on your suggestions:

- Defaulted package creation to the current working directory.
-  Removed namespace arguments from launch files.
-  Differentiate between startup and reconfigurable parameters in both C++ and Python nodes.
-  Moved method documentation to the header files and removed unnecessary comments.

StefanFabian commented:
Also, when creating CI, it creates a file called ' .pre-commit-config.yaml ' with a space before and after, which does not seem intentional to me.
